src/Maestro/Microsoft.DotNet.Maestro.Tasks/src/PushMetadataToBuildAssetRegistry.cs

```diff
@@ -42,6 +43,10 @@ public class PushMetadataToBuildAssetRegistry : MSBuild.Task, ICancelableTask
 
         public string AssetVersion { get; set; }
 
+        public string AssetCategory { get; set; }
+
+        public ConditionalWeakTable<AssetData, string> assetWithCategory = new ConditionalWeakTable<AssetData, string>();
```

Comment:
Should this be public?

Reply:
Also, what's the reasoning for using a `ConditionalWeakTable` vs. a normal dictionary?

Reply:
Also, can you add some comments here as to why we're using a side-dictionary to maintain the category? I think this solution is okay, though I think it would be 'cleaner' to have the flow be the following:

1. The assets, signing info are merged, etc. are merged using the **original** types (PackageArtifactModel/BlobArtifactModel).
2. The merged xml manifest is created and pushed to artifacts
3. The Maestro build model is created using this merged data.

That would eliminate the need for side-table information and make the implementation cleaner, though it would be a decent amount of additional code change. Right now the use of AssetData and BuildData is a bit odd. We're basically starting as a model based on the xml manifest, constructing the merged manifest using the Maestro model of build data, then attempting to go back to the model based on the xml manifest. which is a little awkward. It worked fine because the models were close enough to each other to work in most scenarios, but it wasn't ideal.

Reply:
I was slightly hesitant to change the current functionality, so I added a the fix with the table. But I don't want to add more technical debt, so I will try an clean this up. Thanks for clarifying this!